src/xds.rs

```diff
@@ -150,6 +150,9 @@ impl ProxyStateUpdater {
                     .remove_endpoint(&prev.uid, &endpoint_uid(&prev.uid, None));
             }
 
+            // clear the corresponding identity from SecretManager
+            self.cert_fetcher.clear_cert(&prev.identity());
```

Comment:
ok, calling the clear in the remove() function sounds like a no-go, since multiple pods/workloads can use the same identity/service account.  I think it is still safe in the  `insert_workload()` since an update would result in a delete immediately followed by a prefect_cert.

Reply:
in addition to delete, I am also thinking of a update as well.  In the event of a root cert rotation, there is a full push, where all the identities on the node need to be updated which is why I am thinking of the `insert` case as well

Reply:
example flow on a Full Push:
```
2023-11-17T19:48:58.633342Z  INFO xds{id=2}: ztunnel::xds::client: received response type_url="type.googleapis.com/istio.workload.Address" size=15
2023-11-17T19:48:58.634024Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/istio-system/istiod-5fc86c58d7-6zhjk
2023-11-17T19:48:58.634266Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/istio-system/istiod-5fc86c58d7-6zhjk
2023-11-17T19:48:58.634654Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/kube-system/coredns-5d78c9869d-lfhbw
2023-11-17T19:48:58.634861Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/kube-system/coredns-5d78c9869d-lfhbw
2023-11-17T19:48:58.635178Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/local-path-storage/local-path-provisioner-6bc4bddd6b-zktwb
2023-11-17T19:48:58.635393Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/local-path-storage/local-path-provisioner-6bc4bddd6b-zktwb
2023-11-17T19:48:58.635580Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/metallb-system/controller-574698dfb9-8bpgc
2023-11-17T19:48:58.635667Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/metallb-system/controller-574698dfb9-8bpgc
2023-11-17T19:48:58.635774Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/istio-system/ztunnel-pjz4l
2023-11-17T19:48:58.635835Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/istio-system/ztunnel-pjz4l
2023-11-17T19:48:58.635938Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/istio-system/ztunnel-24rvb
2023-11-17T19:48:58.636008Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/istio-system/ztunnel-24rvb
2023-11-17T19:48:58.636110Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/kube-system/coredns-5d78c9869d-nhbj8
2023-11-17T19:48:58.636263Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/kube-system/coredns-5d78c9869d-nhbj8
2023-11-17T19:48:58.637799Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/kube-system/metrics-server-699dbf548-97d54
2023-11-17T19:48:58.637915Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/kube-system/metrics-server-699dbf548-97d54
2023-11-17T19:48:58.638154Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/istio-system/ztunnel-bzsz5
2023-11-17T19:48:58.638243Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/istio-system/ztunnel-bzsz5
2023-11-17T19:48:58.638353Z DEBUG xds{id=2}: ztunnel::xds: handling insert Kubernetes//Pod/default/sleep-9454cc476-xmbq7
2023-11-17T19:48:58.638423Z DEBUG xds{id=2}: ztunnel::xds: handling delete Kubernetes//Pod/default/sleep-9454cc476-xmbq7
2023-11-17T19:48:58.639160Z DEBUG xds{id=2}: ztunnel::xds::client: sending response type_url="type.googleapis.com/istio.workload.Address" nonce="208bbcbe-7874-46a9-b9ff-c63767938a97" type=Ack
2023-11-17T19:48:58.639407Z DEBUG ztunnel::xds::client: sending request type_url="type.googleapis.com/istio.security.Authorization"
2023-11-17T19:48:58.639609Z DEBUG ztunnel::xds::client: sending request type_url="type.googleapis.com/istio.workload.Address"
2023-11-17T19:48:58.639763Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/istio-system/sa/istiod"
2023-11-17T19:48:58.639834Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/kube-system/sa/coredns"
2023-11-17T19:48:58.639885Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/local-path-storage/sa/local-path-provisioner-service-account"
2023-11-17T19:48:58.639936Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/metallb-system/sa/controller"
2023-11-17T19:48:58.639985Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/istio-system/sa/ztunnel"
2023-11-17T19:48:58.640030Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/istio-system/sa/ztunnel"
2023-11-17T19:48:58.640071Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/kube-system/sa/coredns"
2023-11-17T19:48:58.640116Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/kube-system/sa/metrics-server"
2023-11-17T19:48:58.640148Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/istio-system/sa/ztunnel"
2023-11-17T19:48:58.640178Z DEBUG ztunnel::identity::manager: gihanson cleared identity "spiffe://cluster.local/ns/default/sa/sleep"
2023-11-17T19:48:58.640323Z DEBUG ztunnel::identity::manager: gihanson new identity "spiffe://cluster.local/ns/default/sa/sleep"
```